File: Frontend/demo_app/lib/models/quiz.dart
class Quiz {
  final int id;
  final String question;
  final String optionA;
  final String optionB;
  final String optionC;
  final String optionD;
  final String correctAnswer;
  final String topic;
  final String level;
  final String? feedback;
  final DateTime? createdAt;

  Quiz({
    required this.id,
    required this.question,
    required this.optionA,
    required this.optionB,
    required this.optionC,
    required this.optionD,
    required this.correctAnswer,
    required this.topic,
    required this.level,
    this.createdAt,
    this.feedback,
  });

  factory Quiz.fromJson(Map<String, dynamic> json) {
    return Quiz(
      id: json['id'],
      question: json['question'],
      optionA: json['optionA'],
      optionB: json['optionB'],
      optionC: json['optionC'],
      optionD: json['optionD'],
      correctAnswer: json['correctAnswer'],
      topic: json['topic'],
      level: json['level'],
      createdAt:
          json['createdAt'] != null ? DateTime.parse(json['createdAt']) : null,
      feedback: json['feedback'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'question': question,
      'optionA': optionA,
      'optionB': optionB,
      'optionC': optionC,
      'optionD': optionD,
      'correctAnswer': correctAnswer,
      'topic': topic,
      'level': level,
      'createdAt': createdAt?.toIso8601String(),
      'feedback': feedback,
    };
  }

  List<String> get options => [optionA, optionB, optionC, optionD];

  Map<String, String> get optionsMap => {
    'A': optionA,
    'B': optionB,
    'C': optionC,
    'D': optionD,
  };
}

File: Backend/musicquiz/src/main/java/com/example/musicquiz/model/Quiz.java
package com.example.musicquiz.model;

import jakarta.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "quizzes")
public class Quiz {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    // Keep ALL your existing fields exactly as they are
    private String question;
    private String optionA;
    private String optionB;
    private String optionC;
    private String optionD;
    private String correctAnswer;

    @Column(columnDefinition = "TEXT")
    private String feedback;

    public String getFeedback() {
        return feedback;
    }

    public void setFeedback(String feedback) {
        this.feedback = feedback;
    }

    @Column(nullable = false)
    private String topic;

    private String level; // e.g., Beginner, Intermediate, Expert

    // Just add this one field for basic tracking
    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }

    // Keep ALL your existing constructors exactly as they are
    public Quiz() {}

    public Quiz(String question, String optionA, String optionB, String optionC, String optionD, String correctAnswer, String level) {
        this.question = question;
        this.optionA = optionA;
        this.optionB = optionB;
        this.optionC = optionC;
        this.optionD = optionD;
        this.correctAnswer = correctAnswer;
        this.level = level;
    }

    // Keep ALL your existing getters and setters exactly as they are
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public String getQuestion() { return question; }
    public void setQuestion(String question) { this.question = question; }

    public String getOptionA() { return optionA; }
    public void setOptionA(String optionA) { this.optionA = optionA; }

    public String getOptionB() { return optionB; }
    public void setOptionB(String optionB) { this.optionB = optionB; }

    public String getOptionC() { return optionC; }
    public void setOptionC(String optionC) { this.optionC = optionC; }

    public String getOptionD() { return optionD; }
    public void setOptionD(String optionD) { this.optionD = optionD; }

    public String getCorrectAnswer() { return correctAnswer; }
    public void setCorrectAnswer(String correctAnswer) { this.correctAnswer = correctAnswer; }

    public String getTopic() { return topic; }
    public void setTopic(String topic) { this.topic = topic; }

    public String getLevel() { return level; }
    public void setLevel(String level) { this.level = level; }

    // Only add getter/setter for the new field
    public LocalDateTime getCreatedAt() { return createdAt; }
    public void setCreatedAt(LocalDateTime createdAt) { this.createdAt = createdAt; }
}